app/Controller/AppController.php

```diff
@@ -1101,6 +1101,7 @@ private function __customAuthentication($server)
                 if ($user['User']) {
                     unset($user['User']['gpgkey']);
                     unset($user['User']['certif_public']);
+                    $this->User->updateLoginTimes($user['User']);
```

Comment:
I believe, but am not 100% certain, this will update the login times for every page request. Ideally the server should do the update only on a transition from unauthenticated to authenticated.  But detecting that transition in a stateless server is difficult!

Reply:
This should be OK, because after login, user has session stored in cookie. So this login process is repeated just after session expires.

Reply:
Ok so that's how you detect the transition - if there's no MISP cookie, call login which sets the cookie AND updates the current & last login times, do I have that correct?

Reply:
Yes, thats correct.